mne/io/nicolet/nicolet.py

```diff
@@ -0,0 +1,210 @@
+        Names of channels or list of indices that should be designated
+        EOG channels. Values should correspond to the electrodes in the
+        data file. Default is None.
+    ecg : list or tuple
+        Names of channels or list of indices that should be designated
+        ECG channels. Values should correspond to the electrodes in the
+        data file. Default is None.
+    emg : list or tuple
+        Names of channels or list of indices that should be designated
+        EMG channels. If None (default), the channel names beginning with
+        ``EMG`` are used.
+    misc : list or tuple
+        Names of channels or list of indices that should be designated
+        MISC channels. Values should correspond to the electrodes in the
+        data file. Default is None.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    def __init__(self, input_fname, montage=None, eog=None, ecg=None, emg=None,
+                 misc=None, preload=False, verbose=None):
+        input_fname = path.abspath(input_fname)
+        info, header_info = _get_nicolet_info(input_fname, eog, ecg, emg, misc)
+        last_samps = [header_info['num_samples']]
+        _check_update_montage(info, montage)
+        super(RawNicolet, self).__init__(
+            info, preload, filenames=[input_fname], raw_extras=[header_info],
+            last_samps=last_samps, orig_format='int',
+            verbose=verbose)
+
+    def _read_segment_file(self, data, idx, offset, fi, start, stop, cals,
+                           mult):
+        """Read a chunk of raw data"""
+        if mult is not None:
+            raise NotImplementedError()
+        sel = np.arange(self.info['nchan'])[idx]
+
+        cal = np.array([ch['cal'] for ch in self.info['chs']])
+
+        data = data[:, offset:offset + (stop + 1 - start)]
+        data_offset = self.info['nchan'] * start * 2
+        with open(self._filenames[fi], 'rb', buffering=0) as fid:
+            # extract data
+            fid.seek(data_offset)
+            for bi in range(stop - start):
+                for ch_idx in range(self.info['nchan']):
+                    if ch_idx not in sel:
+                        fid.seek(2, 1)
+                        continue
+                    else:
+                        byte = fid.read(2)
+                        this_data = np.fromstring(byte, '<i2')[0]
+                        data[ch_idx, bi] = this_data * cal[ch_idx]
```

Comment:
looking at it a second time I suspect this for loop over samples is terribly slow. I think it's way faster to read the full buffer and then only select the channels in the idx list. If it's too big of a memory overhead then branch with an "if" depending if a low number of channels is requested or most of it.

please tell me how long it takes to read a 50MB.

it should take less than 1s